app/models/elastic_search_recommended_link.rb

```diff
@@ -0,0 +1,60 @@
+class ElasticSearchRecommendedLink
+  # _id and _type fields should be included in the body when
+  # being sent via Rummager but not when being sent directly to ElasticSearch.
+  #
+  # TODO: Alter Rummager's endpoint to take an optional id and type param
+  # in place of this functionality.
+  def initialize(recommended_link, include_id_and_type_in_body: false)
+    @recommended_link = recommended_link
+    @include_id_and_type_in_body = include_id_and_type_in_body
+  end
+
+  def header
+    {
+      index: {
+        _id: id,
+        _type: type
+      }
+    }
+  end
+
+  def body
+    es_body = {
+      link: link,
+      details: details.to_json
+    }
+
+    if @include_id_and_type_in_body
```

Comment:
If we're doing a conditional on an instance var here, let's set up an `attr_reader` and use that instead. Offers some protection against subtle naming errors as this code changes. Given that, you may want to set up attr_readers for both values in your initializer and use them throughout, for consistency.